### PR TITLE
Made function download_image

### PR DESCRIPTION
**Pull request by MrRaccoonUA**
# What does this pull request do?
In this PR we download images with the desired resolution and the desired name

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -3,10 +3,20 @@
 import time
 import logging
 import re
+import os
+import urllib.request
+from PIL import Image
 from collections import namedtuple
 
+
+if not os.path.exists("log"):
```
Comment (aartiukh) [inline main.py:12]: I'd suggest to have one `out` directory

File: main.py [hunk 2]
```diff
@@ -26,8 +36,9 @@ def api_search(upc, API_URL):
     }
     resp = requests.get(API_URL + upc, headers=headers)
     data = json.loads(resp.text)
-    logger.info(f'UPC: {upc} - Status: {data["code"]}')
-    if data['code'] != 'OK':
+    logger.info(f'UPC: {upc} - API Status: {data["code"]}')
+    if data['code'] != 'OK' or len(data['items']) == 0:
+        logger.info(f'UPC: {upc} - API ERROR Status: {data["code"]}')
```
Comment (aartiukh) [inline main.py:41]:
```suggestion
        logger.error(f'UPC: {upc} - API ERROR Status: {data["code"]}')
```

File: main.py [hunk 3]
```diff
@@ -79,20 +90,37 @@ def walmart_extractor(walmart):
...
     delete_spaces = character_search.replace(' ', '_')
     delete_double_underscore = delete_spaces.replace('__', '_')
-    log_upc = upc + ' - ' + delete_double_underscore
-    logger.info(f'Rename file {log_upc}: OK')
+    logger.info(f'UPC: {upc} - Rename {delete_double_underscore}: OK')
 
     return File_Name(delete_double_underscore)
 
 
+def download_image(upc, link, name, width_dimension, height_dimension):
```
Comment (aartiukh) [inline main.py:106]: Add documentation

File: main.py [hunk 4]
```diff
@@ -79,20 +90,37 @@ def walmart_extractor(walmart):
...
+
+    def get_image_from_url(url, dir_path, full_name):
+        full_path = dir_path + full_name + '.jpg'
```
Comment (aartiukh) [inline main.py:109]: use `os.path.join`

File: main.py [hunk 5]
```diff
@@ -79,20 +90,37 @@ def walmart_extractor(walmart):
...
+        urllib.request.urlretrieve(url, full_path)
+        urllib.request.urlcleanup()
+
+    im = Image.open(urllib.request.urlopen(link))
+    width, height = im.size
+
+    if width >= width_dimension or height >= height_dimension:
+        get_image_from_url(link, 'images/', name)
+        logger.info(f'UPC: {upc} - Download Status: OK')
+        im.close()
```
Comment (aartiukh) [inline main.py:119]: call `im.close` below, not only in this condition

File: main.py [hunk 6]
```diff
@@ -102,20 +130,25 @@ def image_name(upc, product_name):
         ele = int(input())
         upc_list.append(ele)
     upc_list = [str(x) for x in upc_list]
+    print('Start searching...')
```
Comment (aartiukh) [inline main.py:133]: log

File: main.py [hunk 7]
```diff
@@ -102,20 +130,25 @@ def image_name(upc, product_name):
...
 
     API_URL_GET = 'https://api.upcitemdb.com/prod/trial/lookup?upc='
 
     for code in upc_list:
-        start_time = time.time()
+
+        API_COOL_DOWN_TIMEOUT = 10
```
Comment (aartiukh) [inline main.py:139]:
```suggestion
        API_COOL_DOWN_TIMEOUT_SECS = 10
```

File: main.py [hunk 8]
```diff
@@ -102,20 +130,25 @@ def image_name(upc, product_name):
...
 
-        API_COOL_DOWN_TIMEOUT = 10
-        iteration_time = time.time() - start_time
-        cool_down = API_COOL_DOWN_TIMEOUT - iteration_time
-        if len(upc_list) > 1 and code != upc_list[-1]:
-            time.sleep(cool_down)
+        WIDTH_DIMENSION = 600
+        HEIGHT_DIMENSION = 600
+
+        download_image(detail.UPC, extracted_link.resized_link, file_name.file_name, WIDTH_DIMENSION, HEIGHT_DIMENSION)
```
Comment (aartiukh) [inline main.py:154]: `file_name.file_name` looks weird

**Review by aartiukh — CHANGES_REQUESTED** [review]

File: main.py (hunk 8 above)
Comment (MrRaccoonUA) [reply main.py]: Fixed

File: main.py (hunk 7 above)
Comment (MrRaccoonUA) [reply main.py]: Fixed

File: main.py (hunk 6 above)
Comment (MrRaccoonUA) [reply main.py]: Fixed

File: main.py (hunk 5 above)
Comment (MrRaccoonUA) [reply main.py]: Fixed

File: main.py (hunk 4 above)
Comment (MrRaccoonUA) [reply main.py]: Done

File: main.py (hunk 3 above)
Comment (MrRaccoonUA) [reply main.py:102]: Done

File: main.py (hunk 2 above)
Comment (MrRaccoonUA) [reply main.py]: Fixed

File: main.py (hunk 1 above)
Comment (MrRaccoonUA) [reply main.py]: Done

**Review by aartiukh — APPROVED** [review]
Well done 👍

**Pull request merged**